multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -270,6 +284,55 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(commonArea
 	return nil
 }
 
+func (c *StaleResCleanupController) cleanupMemberClusterAnnounce() error {
+	clusterSetList := &mcsv1alpha1.ClusterSetList{}
+	if err := c.List(ctx, clusterSetList, &client.ListOptions{}); err != nil {
+		return err
+	}
+	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
+	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+		return err
+	}
+
+	for _, memberClusterAnnounce := range memberClusterAnnounceList.Items {
+		var exist bool
+		var needToDelete bool
+		if len(clusterSetList.Items) != 0 {
+			members := clusterSetList.Items[0].Spec.Members
+			for _, member := range members {
+				if member.ClusterID == memberClusterAnnounce.ClusterID {
+					exist = true
+					break
+				}
+			}
+			if !exist {
+				klog.InfoS("Cleaning up stale MemberClusterAnnounce. It does exist in the member cluster list", "MemberClusterAnnounce", klog.KObj(&memberClusterAnnounce))
```

Comment:
You shouldn't use `&memberClusterAnnounce` here, you need a line to assign it to a new variable and use it in the loop. You may google why it's needed.

Reply:
I knew in the `for...each` range. The value always uses the same address. But the process is serial, maybe using the same address is okay? I'm not sure, added a new variable to avoid the risk.